tensorflow/tools/ci_build/builds/check_system_libs.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+# Checks that the options mentioned in syslibs_configure.bzl are consistent with the valid options in workspace.bzl
+# Expects the tensorflow source folder as the first argument
+
+import sys
+import os
+from glob import glob
+
+tf_source_path = sys.argv[1]
+
+if not os.path.isdir(tf_source_path):
+  raise ValueError('The path to the TensorFlow source must be passed as'
+                   ' the first argument')
+
+syslibs_configure_path = os.path.join(tf_source_path, 'third_party',
+                                      'systemlibs', 'syslibs_configure.bzl')
+workspace_path = os.path.join(tf_source_path, 'tensorflow', 'workspace.bzl')
+third_party_path = os.path.join(tf_source_path, 'third_party')
+third_party_glob = os.path.join(third_party_path, '*', 'workspace.bzl')
+
+# Stub only
+def repository_rule(**kwargs):
+  del kwargs
+
+# Populates VALID_LIBS
+with open(syslibs_configure_path, 'r') as f:
+  exec(f.read())
+syslibs = set(VALID_LIBS)
+
+syslibs_from_workspace = set()
+
+def extract_system_builds(filepath):
+  current_name = None
+  with open(filepath, 'r') as f:
+    for line in f:
+      line = line.strip()
+      if line.startswith('name = '):
+        current_name = line[7:-1].strip('"')
+      elif line.startswith('system_build_file = '):
```

Comment:
How complicated would it be to also check that the file exists? (ie to detect the recent typing_extensions missing file)

Reply:
Added. Current output:
```
Missing system build files: third_party/systemlibs/typing_extensions.BUILD
Libs missing from syslibs_configure: typing_extensions_archive
```